lib/transports/odbcTransport.js

```diff
@@ -51,36 +52,52 @@ function odbcCall(config, xmlInput, done) {
     console.log(`SQL to run is ${sql}`);
   }
 
-  odbc.connect(connectionString, (connectError, connection) => {
-    if (connectError) {
-      done(connectError, null);
+  function processResults(results) {
+    if (!results) {
+      done('Empty result set was returned', null);
       return;
     }
+
+    let xmlOutput = '';
+
+    results.forEach((chunk) => {
+      xmlOutput += chunk.OUT151;
+    });
+    done(null, xmlOutput);
+  }
+
+  function query(connection) {
     connection.query(sql, [ipc, ctl, xmlInput], (queryError, results) => {
       if (queryError) {
         done(queryError, null);
         return;
       }
-      connection.close((closeError) => {
-        if (closeError) {
-          done(closeError, null);
-          return;
-        }
-
-        if (!results) {
-          done('Empty result set was returned', null);
-          return;
-        }
+      if (!odbcConnection) {
+        connection.close((closeError) => {
+          if (closeError) {
+            done(closeError, null);
+            return;
+          }
 
-        let xmlOutput = '';
-
-        results.forEach((chunk) => {
-          xmlOutput += chunk.OUT151;
+          processResults(results);
         });
-        done(null, xmlOutput);
-      });
+      } else {
+        processResults(results);
+      }
     });
-  });
+  }
```

Comment:
@ehenson

I think this would be great to get this merged in. I would like to request some changes before merging this one in though.

Instead of nesting the [processResults](https://github.com/IBM/nodejs-itoolkit/pull/282/files#diff-edd0e41eebd87a05c24851a8eda588a00a33d51b003a188a49e22326f0a0c777R55) and [query](https://github.com/IBM/nodejs-itoolkit/pull/282/files#diff-edd0e41eebd87a05c24851a8eda588a00a33d51b003a188a49e22326f0a0c777R69) functions maybe this should be split into aone method because whether or not an existing connection is passed we would need to query xml service and process the results. The only difference would be if we should close the connection or not. If an existing connection was not passed close it otherwise don't call close.

Checkout this [branch](https://github.com/abmusse/nodejs-itoolkit/blob/8cb61814cf782d630e9e2488f991c9bcddcfea9f/lib/transports/odbcTransport.js#L34-L134) with how I think this could be done

Let me know If you have time to add the requested changes so we can fix this up otherwise I can pull these changes from this [branch](https://github.com/abmusse/nodejs-itoolkit/tree/odbc-transport-use-existing-connection)  into a new PR and work on getting it merged.

Reply:
@abmusse 
As much as I would like to assist, I cannot get to this any time soon.  Sorry!
You are correct about "if we should close the connection or not." 
Will the consuming application need to pass the `shouldCloseConnection` parameter instead of the code checking for a null/undefined connection object?

Reply:
We could make this a configurable transport option. That way the user could allow closing the connection object if desired.  Would this be desirable for your application?

Reply:
I'm flexible either way as long as a programmer is fully aware that it will close the connection on their behalf unless the parameter is set appropriately.  In my case, I use the same connection to query results of a program call in the QTEMP library, and if it's closed I'll lose the reference to the job.

Reply:
Oh Ok as written now, the consuming application would not need to pass `shouldCloseConnection`. If a `odbcConnection` option is set/passed by the user `shouldCloseConnection` is set to false therefore not closing the passed connection. If a connection was not passed then we create one implicitly and run the query, process results, close out that connection.

Reply:
I like it!  Thanks!